integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -0,0 +1,579 @@
+        init_params = data.get("init_parameters", {})
+        if "streaming_callback" in init_params and init_params["streaming_callback"] is not None:
+            init_params["streaming_callback"] = deserialize_callable(init_params["streaming_callback"])
+        return default_from_dict(cls, data)
+
+    def _handle_streaming_response_direct(
+        self, response_stream, streaming_callback: StreamingCallbackT
+    ) -> Dict[str, Any]:
+        """
+        Handle streaming response from Google Gen AI generate_content_stream.
+
+        :param response_stream: The streaming response from generate_content_stream.
+        :param streaming_callback: The callback function for streaming chunks.
+        :returns: A dictionary with the replies.
+        """
+        try:
+            # Collect all chunks to build the final response
+            all_text_parts = []
+            all_tool_calls = []
+            final_finish_reason = None
+
+            for chunk in response_stream:
+                # Each chunk is a GenerateContentResponse
+                chunk_text = ""
+
+                # Process candidates in the chunk
+                if chunk.candidates:
+                    candidate = chunk.candidates[0]  # Take the first candidate
+
+                    # Extract content from parts
+                    if candidate.content and candidate.content.parts:
+                        for part in candidate.content.parts:
+                            if part.text:
+                                chunk_text += part.text
+                                all_text_parts.append(part.text)
+                            elif part.function_call:
+                                tool_call = ToolCall(
+                                    tool_name=part.function_call.name,
+                                    arguments=dict(part.function_call.args) if part.function_call.args else {},
+                                    id=part.function_call.id,
+                                )
+                                all_tool_calls.append(tool_call)
+
+                    # Update finish reason (keep the last non-None value)
+                    if candidate.finish_reason:
+                        final_finish_reason = candidate.finish_reason
+
+                # Send streaming chunk if there's text content
+                if chunk_text:
+                    streaming_chunk = StreamingChunk(content=chunk_text, meta={"chunk": chunk})
+                    streaming_callback(streaming_chunk)
+
+            # Build final response from collected data
+            final_text = "".join(all_text_parts)
+            message = ChatMessage.from_assistant(text=final_text, tool_calls=all_tool_calls)
+
+            # Add metadata
+            message._meta.update(
+                {
+                    "model": self._model,
+                    "finish_reason": str(final_finish_reason) if final_finish_reason else None,
+                }
+            )
```

Comment:
Let's just directly write to meta when creating the chat message
```suggestion
            message = ChatMessage.from_assistant(
                text=final_text,
                tool_calls=all_tool_calls,
                meta={
                    "model": self._model,
                    "finish_reason": str(final_finish_reason) if final_finish_reason else None,
                }
            )
```

Reply:
I also want to add `completion_start_time` and proper `usage` to meta so we can have valuable info in tracing